src/Compiler/FSComp.txt

```diff
@@ -679,7 +679,7 @@ tcUnnamedArgumentsDoNotFormPrefix,"The unnamed arguments do not form a prefix of
 824,tcAttributesAreNotPermittedOnLetBindings,"Attributes are not permitted on 'let' bindings in expressions"
 825,tcDefaultValueAttributeRequiresVal,"The 'DefaultValue' attribute may only be used on 'val' declarations"
 826,tcConditionalAttributeRequiresMembers,"The 'ConditionalAttribute' attribute may only be used on members"
-827,tcInvalidActivePatternName,"This is not a valid name for an active pattern"
+827,tcInvalidActivePatternName,"'%s' is not a valid method name. Active patterns may only be defined as let-bound module or class functions."
```

Comment:
I don't think "Active patterns may only be defined as let-bound module or class functions" is correct,[^1] but it seems reasonable to guide the user towards what they probably want to do (use some kind of `let`-binding):

```suggestion
827,tcInvalidActivePatternName,"'%s' is not a valid method name. Use a let-binding instead."
```

[^1]:

    Local (not module-scoped) active patterns are common:
    
    ```fsharp
    let f g x =
        let (|A|B|) y = …
    
        match g x with
        | A -> …
        | B -> …
    ```
    
    Active patterns can also be bound anywhere named patterns are allowed, including function parameters:
    
    ```fsharp
    let f (|A|B|) x = match x with A -> … | B -> …
    ```
    
    ```fsharp
    fun (|A|B|) x -> match x with A -> … | B -> …
    ```
    
    ```fsharp
    let f g x =
        match g with
        | (|A|B|) -> match x with A -> … | B -> …
    ```

Reply:
Sound good. Thanks